README.md
# Description
This cookbook will install SAP Business One for SAP HANA.
It contains the following recipes:
* b1::server
Recipe installs the SAP Business One server (SBO-Common, Server Tools, etc.).
* b1::client
Recipe installs the SAP Business One client.

SAP Business One Partners can find tutorials on how to use these cookbooks and setup Business One test environments with Chef on the [SAP PartnerEdge](https://partneredge.sap.com/en/products/business-one/support.html).

# Limitations

**_The SAP Business One Cookbooks do not follow the Business One installation process as described in the Business One Admin Guide.
It is recommended to use these Chef Cookbooks to only create SAP Business One Test and Demo environments. 
The SAP Business One Support Organization will not support Productive environment created with these Chef Cookbooks._**

* B1 versions from 9.2 Patch 4 are supported. The default version is the 'B1 9.2 PL5'.

# Requirements

Supported B1 Server platforms:

* SUSE 

The b1::server cookbook requires a HANA Server, HANA Client and AFL to be already installed.

Supported B1 Client platforms:

* Windows 

SAP HANA server must be installed first.

# Download and Installation

## Download

This Cookbook is meant to be used in conjonction with a Chef server.
You can use Knife or Berkshelf for example to add this Cookbook to a Chef Repository.

For example, you can add this line to the Berksfile of your wrapper cookbook:

	source 'https://supermarket.chef.io'
	metadata
	cookbook 'b1', git: 'https://github.com/SAP/business-one-hana-chef-cookbook.git'


## Usage 

#### b1h::server
This will install the B1 Server components, including the B1 Server, SLD, License Server, etc. You need to define the archive location, the version and the passwords.



`recipe[b1h::server]`

Attributes: 

	{
		"b1h":{
				"archiverepo": "http://myfileserver/b1-software/hana",
				"archive": "B1H92_PL05.zip",
				"targetdbversion": "920150",
				"targetpatch": "5",
				"systempassword": "MyPassword1",
				"siteuserpassword": "MyPassword1"
		}
	}


#### b1h::client
This will install the B1 Client. You need to define the archive location, the version and the passwords.

`recipe[b1h::client]`

Attributes: 

	{	
		"b1h":{
				"archiverepo": "http://myfileserver/b1-software/hana",
				"archive": "B1H92_PL05.zip"
		}
		"b1hclient":{
				"platform": "x86",
				"dbversion": "920150",
				"patch": "5"
		}
	}
	

# Configuration

## List of cookbook attributes
| Key | Type | Description | Default |
| --------------------------------- | ---- | ----------- | ------- |
| ['b1h']['archiverepo'] | String | **[Required]** Location of the compressed B1 installation archive | nil |
| ['b1h']['archive'] | String | Name of the B1 installation archive file | nil |
| ['b1h']['is_multispan_archive'] | Boolean | Specify if the compressed archive is a multispan archive | false |
| ['b1h']['multispan_archives'] | String array | Names of all the files in the B1 installation multispan archive  | nil |
| ['b1h']['targetdbversion'] | String | **[Required]** B1 Database Version  | nil |
| ['b1h']['targetpatch'] | String | **[Required]** B1 Patch Number | nil |
| ['b1h']['systempassword'] | String | **[Required]** SYSTEM password for the HANA server | nil |
| ['b1h']['siteuserpassword'] | String | **[Required]** B1SiteUser password for the SLD | nil |
| ['b1h']['additional_repo'] | String | Additional RPM Zypper repository | nil |
| ['b1h']['installerfolder'] | String | Temporary folder for the installation files| "C:\\temp\\b1" |
| ['b1hclient']['platform']  | String | Specifies if the 32 or 64 bits version of the client will be installed	| "x64" |
| ['b1hclient']['programfiles']  | String | The installation path of SAP Business One 32 bits | "C:\\Program Files (x86)" |
| ['b1hclient']['programfiles64']  | String | The installation path of SAP Business One 64 bits | "C:\\Program Files" |
| ['b1hclient']['licenseserver'] | String | The SLD server hostname | nil |
| ['b1hclient']['licenseserverip'] | String | The SLD server IP address | nil |
| ['b1hclient']['dbversion'] | String | [Required] For B1 Client. B1 Database Version. e.g. "920120" | nil |
| ['b1hclient']['patch'] | String | [Required] For B1 Client. B1 Patch Number. e.g. "2" | nil |
	
# How to obtain support
This project allows and expects users to post questions or bug reports in the [GitHub bug tracking system](../../issues).

# Contributing
If you would like to contribute, please fork this project and post pull requests.

# License
Copyright (c) 2017 SAP SE or an SAP affiliate company. All rights reserved.
This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the [LICENSE](LICENSE.txt) file.












CHANGELOG.md
b1h CHANGELOG
===================

This file is used to list changes made in each version of the b1h cookbook.

0.1.0
-----
- Paul Feeney - Initial release of b1h

- - -
Check the [Markdown Syntax Guide](http://daringfireball.net/projects/markdown/syntax) for help with Markdown.

The [Github Flavored Markdown page](http://github.github.com/github-flavored-markdown/) describes the differences between markdown on github and standard markdown.


recipes/server.rb
#
# Cookbook Name:: b1h
# Recipe:: server
#

require "net/http"
require "uri"
require "json"

include_recipe "zypper"

####################################################################################################
#  --- VARIABLES ---
#
v_archiverepo           = node['b1h']['archiverepo']
v_is_multispan_archive  = node['b1h']['is_multispan_archive']
v_archivename           = node['b1h']['archive']
v_multispan_archives    = node['b1h']['multispan_archives']
v_archive               = node['b1h']['archive']

v_dbversion             = node[:b1h][:targetdbversion]
v_branch                = node[:b1h][:targetbranch]
v_patch                 = node[:b1h][:targetpatch]
v_changelog             = node[:b1h][:targetchangelog]
v_siteuserpassword      = node[:b1h][:siteuserpassword]
v_systempassword        = node[:b1h][:systempassword]
v_plnumber              = v_patch.sub(/_HF[[:digit:]]/, '').sub(/_CL_[[:digit:]]{7}/, '') # Get patch number without HF
v_mv                    = v_dbversion.to_s[0..1]
v_dbversion_part1       = v_dbversion.to_s[0..2]
v_dbversion_part2       = v_dbversion.to_s[3..5]
v_additional_repo       = node[:b1h][:additional_repo]

v_os_platform           = node['platform']
v_os_platform_version   = node['platform_version']


if v_plnumber.to_i < 10
    # Leading zero
    v_patch_leadingzero = "0#{v_patch}"
    v_plnumber_leadingzero = "0#{v_plnumber}"
else
    v_patch_leadingzero = v_patch
    v_plnumber_leadingzero = v_plnumber
end

if(v_archiverepo == nil)
    if v_branch.downcase == "smp"
        v_archiverepo = node['b1h']['patchrepo']
    elsif v_branch.downcase == "rel"
        v_archiverepo = node['b1h']['relproductcdrepo']
    else
        raise "Unsupported B1 Branch."
    end
end

if !v_is_multispan_archive
    if v_archive == nil
        if v_branch.downcase == "smp"
            v_archivename = "B1H#{v_mv}_PL#{v_patch_leadingzero}"
            v_archiveextension = "zip"
        elsif v_branch.downcase == "rel"
            v_archivename = "Product_#{v_dbversion_part1}.#{v_dbversion_part2}.#{v_plnumber_leadingzero}_CD_#{v_changelog}_HANA" # Archive name does not include any HF# 
            v_archiveextension = "rar"
        else
            raise "Unsupported B1 Branch."
        end
    else
            v_archivename = v_archive.rpartition(".").first 
            v_archiveextension = v_archive.rpartition(".").last
    end
end

if !v_is_multispan_archive
    v_installerlocalfolder = "/usr/sap/installers/#{v_archivename}"
else
    v_multispan_first_archive = v_multispan_archives[0]
    v_multispan_first_archivename = v_multispan_first_archive.split(".").first
    v_installerlocalfolder = "#{v_installerfolder}\\#{v_multispan_first_archivename}"
end

if v_dbversion.to_i >= 920003
    v_installerlocalsubfolder = "#{v_installerlocalfolder}/Packages.Linux/ServerComponents"
    v_installfilename = "install"
else
    v_installerlocalsubfolder = "#{v_installerlocalfolder}"
    v_installfilename = "install.bin"
end
#
#  --- END VARIABLES ---
####################################################################################################

Chef::Log.info "Variable v_additional_repo: #{ v_additional_repo }"

Chef::Log.info "Variable v_dbversion: #{ v_dbversion }"
Chef::Log.info "Variable v_branch: #{ v_branch }"
Chef::Log.info "Variable v_patch: #{ v_patch }"
Chef::Log.info "Variable v_siteuserpassword: #{ v_siteuserpassword }"
Chef::Log.info "Variable v_systempassword: #{ v_systempassword }"

Chef::Log.info "Variable v_is_multispan_archive: #{ v_is_multispan_archive }"
Chef::Log.info "Variable v_multispan_first_archive: #{ v_multispan_first_archive }"
Chef::Log.info "Variable v_multispan_first_archivename: #{ v_multispan_first_archivename }"

Chef::Log.info "Variable v_installerlocalfolder: #{ v_installerlocalfolder }"
Chef::Log.info "Variable v_installerlocalsubfolder: #{ v_installerlocalsubfolder }"
Chef::Log.info "Variable v_installfilename: #{ v_installfilename }"

Chef::Log.info "Variable v_os_platform: #{ v_os_platform }"
Chef::Log.info "Variable v_os_platform_version: #{ v_os_platform_version }"



if v_additional_repo != nil
  zypper_repo 'b1h-additional-repo' do
    repo_name 'b1h-additional-repo'
    uri v_additional_repo
    autorefresh true
  end
end

execute "zypper refresh" do
    command "zypper --non-interactive --no-gpg-checks refresh"
end

directory "#{v_installerlocalfolder}" do
  owner "root"
  group "root"
  mode 0777
  recursive true
  not_if { ::File.exist?("#{v_installerlocalfolder}") }
end

# if copying source file uing chef remote_file resource, the staged file is saved to /tmp which may not have enough capacity. Use wget instead.
if !v_is_multispan_archive
  execute "Copy B1 Archive" do
    command "wget #{v_archiverepo}/#{v_archivename}.#{v_archiveextension} -O #{v_installerlocalfolder}/#{v_archivename}.#{v_archiveextension}"
    not_if { ::File.exists?("#{v_installerlocalfolder}/#{v_archivename}.#{v_archiveextension}") || ::File.exists?("#{v_installerlocalfolder}/UNZIP_COMPLETE")}
  end
else
  v_multispan_archives.each do |file|
    execute "Copy B1 Archive" do
      command "wget #{v_archiverepo}/#{file} -O #{v_installerlocalfolder}/#{file}"
      not_if { ::File.exists?("#{v_installerlocalfolder}/#{file}") || ::File.exists?("#{v_installerlocalfolder}/UNZIP_COMPLETE")}
    end
  end
end

bash 'set permissions of copied folder' do
  cwd ::File.dirname("#{v_installerlocalfolder}")
  code <<-EOH
    chmod -R 777 .
    EOH
  only_if { ::File.exists?("#{v_installerlocalfolder}") }
end

#additinal permissions due to 'b1service0' user change in HANA revision 97
bash 'set permissions of B1 log folder' do
  cwd ::File.dirname("/usr/sap/SAPBusinessOne/logs")
  code <<-EOH
    chmod -R 755 .
    EOH
  only_if { ::File.exists?("/usr/sap/SAPBusinessOne/logs") }
end

package 'p7zip' do
  action :install
end

if !v_is_multispan_archive
  bash 'Uncompress' do
    cwd "#{v_installerlocalfolder}"
    code <<-EOH
      7z x -y #{v_archivename}.#{v_archiveextension}
      EOH
    not_if {::File.exist?("#{v_installerlocalfolder}/UNZIP_COMPLETE")}
  end
else
  bash 'Uncompress' do
    cwd "#{v_installerlocalfolder}"
    code <<-EOH
      7z x -y #{v_multispan_first_archive}
      EOH
    not_if {::File.exist?("#{v_installerlocalfolder}/UNZIP_COMPLETE")}
  end
end

bash 'record unzip complete' do
  cwd "#{v_installerlocalfolder}"
  code <<-EOH
    echo > UNZIP_COMPLETE
  EOH
end

bash 'set permissions for extracted files' do
  cwd ::File.dirname("#{v_installerlocalfolder}")
  code <<-EOH
    chmod -R 777 *
    EOH
  only_if { ::File.exists?("#{v_installerlocalfolder}") }
end

if !v_is_multispan_archive
  bash 'delete compressed archive' do
    code <<-EOH
      rm -f "#{v_installerlocalfolder}\\#{v_archivename}.#{v_archiveextension}"
    EOH
    only_if { ::File.exists?("#{v_installerlocalfolder}\\#{v_archivename}.#{v_archiveextension}")}
  end
else
  v_multispan_archives.each do |file|
    bash 'delete compressed archive' do
      code <<-EOH
        rm -f "#{v_installerlocalfolder}\\#{file}"
      EOH
      only_if { ::File.exists?("#{v_installerlocalfolder}\\#{file}")}
    end
  end
end

#Add silent upgrade config file
if v_dbversion.to_i >= 930000 
  template "#{v_installerlocalsubfolder}/silentInstallConfig" do
    source "silentInstallConfig-9.3PL0.erb"
    variables(
      :host => node['fqdn'],
      :v_siteuserpassword => v_siteuserpassword,
      :v_systempassword => v_systempassword,
      :v_installerlocalfolder => v_installerlocalfolder
    )
    mode 0777
  end
elsif v_dbversion.to_i >= 920003 
  template "#{v_installerlocalsubfolder}/silentInstallConfig" do
    source "silentInstallConfig-9.2PL0.erb"
    variables(
      :host => node['fqdn'],
      :v_siteuserpassword => v_siteuserpassword,
      :v_systempassword => v_systempassword
    )
    mode 0777
  end
else
  block do
    raise "B1 version #{v_dbversion} not supported"
  end
end

# install samba
package 'samba' do
  action :install
end

#enable samba service
service "smb" do 
  pattern "smb"
  action [:enable, :start]
end

# Prerequisites needed for 9.2 PL3 and up
if v_dbversion.to_i >= 920130 
  # SuSEfirewall2
  bash "install SuSEfirewall2 " do
    user "root"
    code <<-EOH
      zypper install -y SuSEfirewall2
    EOH
    not_if {::File.exist?("/sbin/SuSEfirewall2")}
  end

  package 'nfs-kernel-server' do
    action :install
  end
end

# Prerequisites needed for 9.3 PL0 and up
if v_dbversion.to_i >= 930000 
  package 'python-crypto' do
    action :install
  end

  package 'python-openssl' do
    action :install
  end
end

# Prerequisites needed for 9.2 PL07 and up (see SAP Note 2418458)
# Required versions should be made available yast repositories (e.g. additional_repo) 
if v_os_platform.downcase == "suse" 
  if v_os_platform_version == "11.3" || v_os_platform_version == "11.4"
    if v_dbversion.to_i >= 920170 
      # required libgcc_s1 > 4.8.3
      package 'libgcc_s1' do 
        action :upgrade
      end

      # required libstdc++6 > 4.8.3
      package 'libstdc++6' do
        action :upgrade
      end
    end
  end
end

# Prerequisites needed for SLES12 SP1 (see SAP Note 2458610) 
# Required packages should be made available yast repositories (e.g. additional_repo) 
if v_os_platform.downcase == "suse" 
  if v_os_platform_version == "12.1" 
    package 'rpmbuild' do
      action :install
    end

    package 'compat-libgcrypt11' do
      action :install
    end

    package 'libopenssl0_9_8' do
      action :install
    end
  end
end


#Execute install command
execute "install-b1h" do
  cwd "#{v_installerlocalsubfolder}"
  user "root"
  command "PATH=$PATH:~/bin:.:/sbin/usr/sbin ; ./#{v_installfilename} -i silent -f silentInstallConfig --debug 2> /tmp/b1_server_install_error.log > /tmp/b1_server_install_run.log"
  returns [0, 24, 25]
  action :run
  not_if { ::File.exists?("#{v_installerlocalfolder}/SERVER_INSTALLED_#{v_dbversion}")}
end

template "#{v_installerlocalfolder}/getServerDbVersion.sh" do
  source "getServerDbVersion.sh.erb"
  mode 0777
end

execute "Validate B1 Server Install" do
  cwd "#{v_installerlocalfolder}"
  command "sh ./getServerDbVersion.sh"
  not_if { ::File.exists?("#{v_installerlocalfolder}/SERVER_INSTALLED_#{v_dbversion}")}
end

ruby_block "Validate B1 Server Install" do
  block do
      raise "Server (B1ServerToolsSLD) is not installed or is not on the expected version"
  end
  not_if { ::File.exists?("#{v_installerlocalfolder}/SERVER_INSTALLED_#{v_dbversion}")}
end


recipes/client.rb
#
# Cookbook Name:: b1
# Recipe:: client
#

Chef::Log.info "include seven_zip"
include_recipe "seven_zip"

####################################################################################################
#  --- VARIABLES ---
#
v_archiverepo           = node['b1h']['archiverepo']
v_is_multispan_archive  = node['b1h']['is_multispan_archive']
v_archivename           = node['b1h']['archive']
v_multispan_archives    = node['b1h']['multispan_archives']
v_archive               = node['b1h']['archive']

v_installerfolderbase   = node['b1h']['installerfolder']

v_licenseserver         = node['b1hclient']['licenseserver']
v_licenseserverip       = node['b1hclient']['licenseserverip']

v_dbversion             = node['b1hclient']['dbversion']
v_branch                = node['b1hclient']['branch']
v_patch                 = node['b1hclient']['patch']
v_platform              = node['b1hclient']['platform']
v_changelog             = node['b1hclient']['changelog']

v_plnumber              = v_patch.sub(/_HF[[:digit:]]/, '').sub(/_CL_[[:digit:]]{7}/, '') # Get patch number without HF
v_mv                    = v_dbversion.to_s[0..1]
v_dbversion_part1       = v_dbversion.to_s[0..2]
v_dbversion_part2       = v_dbversion.to_s[3..5]


if v_plnumber.to_i < 10
  # Leading zero
  v_patch_leadingzero = "0#{v_patch}"
  v_plnumber_leadingzero = "0#{v_plnumber}"
else
  v_patch_leadingzero = v_patch
  v_plnumber_leadingzero = v_plnumber
end
  
if(v_archiverepo == nil)
  if v_branch.downcase == "smp"
    v_archiverepo = node['b1h']['patchrepo']
  elsif v_branch.downcase == "rel"
    v_archiverepo = node['b1h']['relproductcdrepo']
  else
    raise "Unsupported B1 Branch."
  end
end

if !v_is_multispan_archive
  if v_archive == nil
    if v_branch.downcase == "smp"
      v_archivename = "B1H#{v_mv}_PL#{v_patch_leadingzero}"
      v_archiveextension        = "zip"
    elsif v_branch.downcase == "rel"
      v_archivename = "Product_#{v_dbversion_part1}.#{v_dbversion_part2}.#{v_plnumber_leadingzero}_CD_#{v_changelog}_HANA" # Archive name does not include any HF# 
      v_archiveextension = "rar"
    else
      raise "Unsupported B1 Branch."
    end
  else
      v_archivename = v_archive.rpartition(".").first 
      v_archiveextension = v_archive.rpartition(".").last
  end
end

if !v_is_multispan_archive
  v_installerfolderextracted = "#{v_installerfolderbase}\\#{v_archivename}"
else
  v_multispan_first_archive = v_multispan_archives[0]
  v_multispan_first_archivename = v_multispan_first_archive.split(".").first
  v_installerfolderextracted = "#{v_installerfolderbase}\\#{v_multispan_first_archivename}"
end

v_platformName          = v_platform == "x86" ? 32 : 64
v_programfiles          = v_platform == "x86" ? node['b1hclient']['programfiles'] : node['b1hclient']['programfiles64']

if v_platform == "x86"
  v_setupExePath = "#{v_installerfolderbase}\\#{v_archivename}\\Packages\\Client\\setup.exe"
  v_setupIssPath = "#{v_installerfolderbase}\\#{v_archivename}\\Packages\\Client\\setup.iss"
else
  v_setupExePath = "#{v_installerfolderbase}\\#{v_archivename}\\Packages.x64\\Client\\setup.exe"
  v_setupIssPath = "#{v_installerfolderbase}\\#{v_archivename}\\Packages.x64\\Client\\setup.iss"
end

v_remotefile            = "#{v_archiverepo}/#{v_archivename}.#{v_archiveextension}"
#
#  --- END VARIABLES ---
####################################################################################################


Chef::Log.info "Variable v_archiverepo: #{v_archiverepo}"
Chef::Log.info "Variable v_licenseserver: #{v_licenseserver}"
Chef::Log.info "Variable v_licenseserverip: #{v_licenseserverip}"
Chef::Log.info "Variable v_dbversion: #{v_dbversion}"
Chef::Log.info "Variable v_branch: #{v_branch}"
Chef::Log.info "Variable v_patch: #{v_patch}"
Chef::Log.info "Variable v_changelog: #{v_changelog}"

Chef::Log.info "Variable v_installerfolderbase: #{v_installerfolderbase}"
Chef::Log.info "Variable v_setupExePath: #{v_setupExePath}"
Chef::Log.info "Variable v_setupIssPath: #{v_setupIssPath}"
Chef::Log.info "Variable v_programfiles: #{v_programfiles}"

Chef::Log.info "Variable v_is_multispan_archive: #{ v_is_multispan_archive }"
Chef::Log.info "Variable v_multispan_first_archive: #{ v_multispan_first_archive }"
Chef::Log.info "Variable v_multispan_first_archivename: #{ v_multispan_first_archivename }"

Chef::Log.info "Variable v_archivename: #{v_archivename}"
Chef::Log.info "Variable v_archiveextension: #{v_archiveextension}"
Chef::Log.info "Variable v_remotefile: #{v_remotefile}"


directory "#{v_installerfolderbase}" do
  action :create
  recursive true
end

if !v_is_multispan_archive
  remote_file 'Copy Patch Archive' do
    path "#{v_installerfolderbase}\\#{v_archivename}.#{v_archiveextension}"
    source "#{v_archiverepo}/#{v_archivename}.#{v_archiveextension}"
    action :create_if_missing
    not_if { ::File.exists?("#{v_installerfolderextracted}\\UNZIP_COMPLETE")}
  end
else
  v_multispan_archives.each do |file|
    remote_file 'Copy Patch Archive' do
      path "#{v_installerfolderbase}\\#{file}"
      source "#{v_archiverepo}/#{file}"
      action :create_if_missing
      not_if { ::File.exists?("#{v_installerfolderextracted}\\UNZIP_COMPLETE")}
    end
  end
end

if !v_is_multispan_archive
  batch 'unzip_installer' do
    code <<-EOH
      7z.exe x #{v_installerfolderbase}\\#{v_archivename}.#{v_archiveextension}  -o#{v_installerfolderextracted} -r -y
    EOH
    not_if { ::File.exists?("#{v_installerfolderextracted}\\UNZIP_COMPLETE")}
  end
else
  batch 'unzip_installer' do
    code <<-EOH
      7z.exe x #{v_installerfolderbase}\\#{v_multispan_first_archive}  -o#{v_installerfolderextracted} -r -y
    EOH
    not_if { ::File.exists?("#{v_installerfolderextracted}\\UNZIP_COMPLETE")}
  end
end

batch 'record unzip complete' do
  cwd "#{v_installerfolderbase}\\#{v_archivename}"
  code <<-EOH
    echo > UNZIP_COMPLETE
    EOH
end

if !v_is_multispan_archive
  batch 'delete installer zip' do
    code <<-EOH
    del "#{v_installerfolderbase}\\#{v_archivename}.#{v_archiveextension}"
    EOH
    only_if { ::File.exists?("#{v_installerfolderbase}\\#{v_archivename}.#{v_archiveextension}")}
  end
else
  v_multispan_archives.each do |file|
    batch 'delete installer zip' do
      code <<-EOH
      del "#{v_installerfolderbase}\\#{file}"
      EOH
      only_if { ::File.exists?("#{v_installerfolderbase}\\#{file}")}
    end
  end
end

Chef::Log.info "Create setup.iss"
template 'setup.iss' do
  path "#{v_setupIssPath}"
  source "setup.iss.erb"
  action :create
end

Chef::Log.info "Install B1 Client"
if v_licenseserver.to_s.strip.empty?
  v_installerZOptions = "/z\"#{v_programfiles}\\SAP\\SAP Business One\""
else
  v_installerZOptions = "/z\"#{v_programfiles}\\SAP\\SAP Business One*#{v_licenseserver}:40000\""
end

# %W[...] is rudy for making an array. Using this method of passing options so that #{v_licenseserver} gets evaluated correctly
#
windows_package 'B1H Client' do
  package_name "SAP Business One Client (#{v_platformName}-bit)"
  source "#{v_setupExePath}"
  #options '/s /z"C:\Program Files (x86)\SAP\SAP Business One*#{v_licenseserver}:40000"'
  options %W[
            /s
            #{v_installerZOptions}
            ].join(' ')

  timeout 2700
  action :install
end

Chef::Log.info "Update b1-local-machine.xml with licenseserver address"

if !v_licenseserver.to_s.strip.empty?

  #Replace "C:\Program Files (x86)\SAP\SAP Business One\Conf\b1-local-machine.xml"
  template 'b1-local-machine.xml' do
     path "#{v_programfiles}\\SAP\\SAP Business One\\Conf\\b1-local-machine.xml"
     source "b1-local-machine.xml.erb"
    variables(
      :v_licenseserver => node[:b1hclient][:licenseserver],
      :v_licenseserverip => node[:b1hclient][:v_licenseserverip]
    )
    action :create
  end

  #Replace "C:\Program Files (x86)\SAP\SAP Business One DI API\Conf\b1-local-machine.xml"
  template 'b1-local-machine.xml' do
      path "#{v_programfiles}\\SAP\\SAP Business One DI API\\Conf\\b1-local-machine.xml"
      source "b1-local-machine.xml.erb"
      variables(
      :v_licenseserver => node[:b1hclient][:licenseserver],
      :v_licenseserverip => node[:b1hclient][:v_licenseserverip]
    )
    action :create
  end
  
end

